twake/frontend/src/app/scenes/Client/ChannelsBar/Modals/SearchListManager.ts

```diff
@@ -0,0 +1,218 @@
+import { ChannelResource } from 'app/models/Channel';
+import { UserType } from 'app/models/User';
+import Strings from 'app/services/utils/strings';
+import UsersService from 'services/user/user.js';
+import Workspaces from 'services/workspaces/workspaces.js';
+import { Collection } from 'services/CollectionsReact/Collections';
+import RouterServices from 'services/RouterService';
+import { getUserParts } from 'app/components/Member/UserParts';
+
+export type GenericChannel = {
+  type: 'user' | 'direct' | 'workspace';
+  sortString: string;
+  filterString: string;
+  lastActivity?: number;
+  resource: UserType | ChannelResource;
+};
+
+class SearchListManager {
+  private workspaceChannels: GenericChannel[];
+  private directChannels: GenericChannel[];
+  private users: GenericChannel[];
+  public list: GenericChannel[];
+
+  constructor() {
+    this.workspaceChannels = [];
+    this.directChannels = [];
+    this.users = [];
+    this.list = [];
+  }
+
+  public bind(search: string): void {
+    const { workspaceId, companyId } = RouterServices.getStateFromRoute();
+    // Path
+    const workspaceChannelsPath = `/channels/v1/companies/${companyId}/workspaces/${workspaceId}/channels/`;
+    const directChannelsPath = `/channels/v1/companies/${companyId}/workspaces/direct/channels/::mine`;
+    const mineWorkspaceChannelsPath = `/channels/v1/companies/${companyId}/workspaces/${workspaceId}/channels/::mine`;
+
+    // Resources
+    const workspaceChannelsResources = this.find({
+      collectionPath: workspaceChannelsPath,
+    }) as ChannelResource[];
+
+    const directChannelsResources = this.find({
+      collectionPath: directChannelsPath,
+    }) as ChannelResource[];
+
+    const mineWorkspaceChannelsResources = this.find({
+      collectionPath: mineWorkspaceChannelsPath,
+    }) as ChannelResource[];
+
+    const usersSearched = this.find({ search }) as UserType[];
+
+    // Filters
+    this.workspaceChannels = this.filterWorkspaceChannels({
+      channels: workspaceChannelsResources,
+      mineWorkspaceChannels: mineWorkspaceChannelsResources,
+    });
+    this.directChannels = this.filterDirectChannels({ channels: directChannelsResources });
+    this.users = this.filterUsers({ users: usersSearched });
+
+    // Concat list
+    this.list = [...this.workspaceChannels, ...this.directChannels, ...this.users];
+
+    this.removeDuplicate();
+
+    this.list = this.list
+      .filter(({ filterString }) => filterString.toUpperCase().indexOf(search.toUpperCase()) > -1)
+      .sort((a, b) => (b.lastActivity || 0) - (a.lastActivity || 0));
+  }
+
+  private find({
+    collectionPath,
+    search,
+  }: {
+    collectionPath?: string;
+    search?: string;
+  }): (UserType | ChannelResource)[] {
+    if (collectionPath?.length) {
+      const collection = Collection.get(collectionPath, ChannelResource);
+      return collection.find({});
```

Comment:
For clarity, you do not have to declare `collection` but just directly return `return Collection.get(args).find({})`;